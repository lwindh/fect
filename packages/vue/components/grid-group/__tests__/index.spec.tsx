import Grid from '../../grid'
import GridGroup from '../index'
import { mount } from '@vue/test-utils'

describe('Grid', () => {
  it('shoud be render as element', () => {
    const wrapper = mount({
      render() {
        return (
          <GridGroup>
            <Grid>T</Grid>
          </GridGroup>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be support breakPoints', () => {
    const wrapper = mount({
      render() {
        return (
          <GridGroup>
            <Grid xs={1} sm={2} md={3} lg={4} xl={5}>
              T
            </Grid>
          </GridGroup>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should be support align', () => {
    const wrapper = mount({
      render() {
        return (
          <GridGroup>
            <Grid justify="flex-end" xl={3}>
              T
            </Grid>
          </GridGroup>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should be supprot use col and count', () => {
    const slots = {
      grid: () => <div>123</div>,
    }
    const wrapper = mount({
      render() {
        return <GridGroup col={4} count={11} v-slots={slots}></GridGroup>
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('props count may be as array', () => {
    const slots = {
      grid: () => <div>123</div>,
    }
    const wrapper = mount({
      render() {
        return <GridGroup col={25} count={[1, 2, 3, 4]} v-slots={slots}></GridGroup>
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
