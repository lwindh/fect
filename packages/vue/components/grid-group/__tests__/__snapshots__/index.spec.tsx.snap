// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid props count may be as array 1`] = `
<div class="fect-grid__group fect-grid__group--wrap " style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start; --fect-grid-gap: calc(0 * 16px * 1/3); --grid-width: 4.166666666666667%;">
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
</div>
`;

exports[`Grid shoud be render as element 1`] = `
<div class="fect-grid__group fect-grid__group--wrap " style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start; --fect-grid-gap: calc(0 * 16px * 1/3);">
  <div class="fect-grid " style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">T</div>
</div>
`;

exports[`Grid should be support align 1`] = `
<div class="fect-grid__group fect-grid__group--wrap " style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start; --fect-grid-gap: calc(0 * 16px * 1/3);">
  <div class="fect-grid fect-grid--xl " style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-end; --xl-grow: 0; --xl-basis: 12.5%; --xl-width: 12.5%;">T</div>
</div>
`;

exports[`Grid should be support breakPoints 1`] = `
<div class="fect-grid__group fect-grid__group--wrap " style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start; --fect-grid-gap: calc(0 * 16px * 1/3);">
  <div class="fect-grid fect-grid--xs fect-grid--sm fect-grid--md fect-grid--lg fect-grid--xl " style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start; --xs-grow: 0; --xs-basis: 4.166666666666667%; --xs-width: 4.166666666666667%; --sm-grow: 0; --sm-basis: 8.333333333333334%; --sm-width: 8.333333333333334%; --md-grow: 0; --md-basis: 12.5%; --md-width: 12.5%; --lg-grow: 0; --lg-basis: 16.666666666666668%; --lg-width: 16.666666666666668%; --xl-grow: 0; --xl-basis: 20.833333333333336%; --xl-width: 20.833333333333336%;">T</div>
</div>
`;

exports[`Grid should be supprot use col and count 1`] = `
<div class="fect-grid__group fect-grid__group--wrap " style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start; --fect-grid-gap: calc(0 * 16px * 1/3); --grid-width: 25%;">
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
  <div class="fect-grid fect-grid--live" style="align-content: flex-start; align-items: stretch; flex-direction: row; justify-content: flex-start;">
    <div>123</div>
  </div>
</div>
`;
